### DISCUSSION TITLE
How to install jsonls pyright using yarn backend instead of npm

### DISCUSSION THREAD

**Discussion by pannet1** [LazyVim/LazyVim#1089] [Q&A] [answered]
Everytime when i start lazyVim Mason tries to load pyright and jsonls with npm. However, i am using yarn package manager and because of this i am getting a warning and failure to install, i guess. i am able to use nvim without any problem but it is annoying to get notified like this everytime.

![image](https://github.com/LazyVim/LazyVim/assets/12350276/44b0865f-9299-4b9d-b1eb-b3aec862c3ae)

can you please help resolve this.

**Comment by jyuan0** [comment]
The easiest thing to do might be to ignore/silence these notifications. See this discussion for an example of how to do that: https://github.com/LazyVim/LazyVim/discussions/830

**Comment by pannet1** [comment]
do you mean to say that if pyright is installed through yarn it is recognized by Lazy and only the notification is showing up initially.

**Reply by dpetka2001** [reply]
You could declare the path to pyright through `cmd` in your lspconfig. Also add to pyright config in your lspconfig `mason = false`, so that mason doesn't try to install it.

**Reply by pannet1** [reply]
@dpetka2001 
if i am not asking too much, can you please help me with the code.

**Reply by dpetka2001** [reply]
Sure do you have your nvim dot files in github available?

**Reply by pannet1** [reply]
yes.

.
├── init.lua
├── lazy-lock.json
├── LICENSE
├── lua
│   ├── config
│   │   ├── autocmds.lua
│   │   ├── keymaps.lua
│   │   ├── lazy.lua
│   │   └── options.lua
│   └── plugins
│       ├── colorscheme.lua
│       ├── example.lua
│       ├── lsp.lua
│       ├── mason.lua
│       ├── mason-nvim-dap.lua
│       ├── mini-sorround.lua
│       ├── nvim-dap.lua
│       ├── treesitter.lua
│       └── which-key.lua
├── README.md
└── stylua.toml

4 directories, 18 files

**Reply by dpetka2001** [reply]
Excuse me but could you give me a link to your nvim dotfiles in github?

**Reply by dpetka2001** [reply]
Something like this in a file in `/lua/plugins/` directory should work
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        pyright = {
          cmd = { "/yarn/full/path/to/pyright-langserver", "--stdio" },
          mason = false,
        },
      },
    },
  },
}
```
or if `pyright-langserver` is in your $PATH then the full-path should not be needed. Just the executable along with `--stdio`.

**Reply by pannet1** [reply]
https://github.com/pannet1/config/tree/main

not forked LazyVim before making my changes. so this contains old nvim.


EDIT: Please ignore this. our replies crossed. will do as you say and revert with the outcome.

**Reply by pannet1** [reply, marked as answer]
before editing

![image](https://github.com/LazyVim/LazyVim/assets/12350276/6aa2c61b-7173-46a6-b8fc-222df765a16d)

after editing

```  -- add pyright to lspconfig
  {
    "neovim/nvim-lspconfig",
    ---@class PluginLspOpts
    opts = {
      ---@type lspconfig.options
      servers = {
        pyright = {
          cmd = { "~/.yarn/bin/pyright-langserver", "--stdio" },
          mason = false,
        },
      },
    },
  },
  -- end return
```

**Reply by dpetka2001** [reply]
Does it work for you? Try with a python file and check `:LspInfo`.

**Reply by pannet1** [reply]
worked like a charm but still notifies :-)

![image](https://github.com/LazyVim/LazyVim/assets/12350276/fb9ca597-792a-4c4a-8c67-245e78fc774f)

**Reply by dpetka2001** [reply]
As far as the red `mason-lspconfig` error is concerned, you should do the same as you did for `pyright`, but this time for `jsonls`. For the yellow warnings, that come presumably from `lspconfig`, you should follow the link #830 that @jyuan0 mentioned already to get rid of them.

**Reply by dpetka2001** [reply]
Actually is the path `~/.yarn/bin` in your $PATH? Because if it is, you could just do `cmd = { "pyright-langserver", "--stdio" }` and potentially not get the yellow warnings at all.

**Reply by dpetka2001** [reply]
Or even delete the whole `cmd` line, since it's the default setting for pyright server either way.

**Reply by pannet1** [reply]
fantastic @dpetka2001 
added yarn binaries to the path and all warnings have gone. i also deleted the whole cmd line :-)
nb: sorry for the late reply. did n't notice your kind help before

**Comment by jyuan0** [comment]
> do you mean to say that if pyright is installed through yarn it is recognized by Lazy and only the notification is showing up initially.

No, but disregard me. The reply from dpetka2001 is better.

**Reply by pannet1** [reply]
@jyuan0 
thank you very much for your dual effort.

**Comment by pannet1** [comment]
> Does it work for you? Try with a python file and check `:LspInfo`.
@dpetka2001 Thank you Iordanis
